cmake-docker-environment/lab-files/targets-example/CMakeLists.txt

```diff
@@ -1,13 +1,12 @@
 cmake_minimum_required(VERSION 3.9.1)
 project(songTest VERSION 1.0.0)
 
-add_library(firstDay SHARED firstDay/firstDay.cpp)
+add_library(firstDay firstDay/firstDay.cpp)
+target_include_directories(firstDay PUBLIC ${CMAKE_CURRENT_LIST_DIR}/firstDay/include)
 
-add_library(secondDay SHARED secondDay/secondDay.cpp)
-target_include_directories(secondDay PUBLIC ${CMAKE_CURRENT_LIST_DIR}/firstDay/include)
-target_link_libraries(secondDay INTERFACE firstDay)
+add_library(secondDay secondDay/secondDay.cpp)
+target_include_directories(secondDay PUBLIC ${CMAKE_CURRENT_LIST_DIR}/secondDay/include)
+target_link_libraries(secondDay PRIVATE firstDay)  # Exercise 2: Make it PUBLIC
 
 add_executable(song main.cpp)
-target_include_directories(song PUBLIC ${CMAKE_CURRENT_LIST_DIR}/secondDay/include/)
-#target_include_directories(song PRIVATE ${CMAKE_CURRENT_LIST_DIR}/firstDay)
-target_link_libraries(song PUBLIC secondDay)
\ No newline at end of file
+target_link_libraries(song secondDay)

```

Comment:
`target_link_libraries` is the function that "connects" one target to another, transferring also information about its includes and so (AFAIK).

In this case, the PUBLIC/PRIVATE/INTERFACE keywords don't make sense as it is an executable (executables are the end application, so they do not propagate symbols in the binary), so it can be omitted.